File: jsimple-json/src/main/java/jsimple/json/readerwriter/JsonBooleanProperty.java
package jsimple.json.readerwriter;

/**
 * @author Bret Johnson
 * @since 5/22/13 9:09 PM
 */
public class JsonBooleanProperty extends JsonProperty {
    public JsonBooleanProperty(String name, int id) {
        super(name, id);
    }

    public boolean readValue(JsonObjectReader objectReader) {
        return (boolean) (Boolean) objectReader.readPropertyValue();
    }

    @Override public Object readValueUntyped(JsonObjectReader objectReader) {
        return objectReader.readPropertyValue();
    }

    public void write(JsonObjectWriter objectWriter, boolean value) {
        objectWriter.writeProperty(this, value);
    }
}

File: jsimple-json/c#/jsimple/json/readerwriter/JsonBooleanProperty.cs
namespace jsimple.json.readerwriter
{

	/// <summary>
	/// @author Bret Johnson
	/// @since 5/22/13 9:09 PM
	/// </summary>
	public class JsonBooleanProperty : JsonProperty
	{
		public JsonBooleanProperty(string name, int id) : base(name, id)
		{
		}

		public virtual bool readValue(JsonObjectReader objectReader)
		{
			return (bool)(bool?) objectReader.readPropertyValue();
		}

		public override object readValueUntyped(JsonObjectReader objectReader)
		{
			return objectReader.readPropertyValue();
		}

		public virtual void write(JsonObjectWriter objectWriter, bool value)
		{
			objectWriter.writeProperty(this, value);
		}
	}

}